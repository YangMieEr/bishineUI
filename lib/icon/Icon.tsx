import * as React from 'react';
import classes from '../util/classes';
import './iconfont.css';
interface IconProps extends React.HTMLAttributes<HTMLElement>{
    name: string;
}

const Icon: React.FunctionComponent<IconProps> = 
({className, name, ...restProps}) => {
    return (
        <i className={classes('iconfont', `icon${name}`, className)}  {...restProps}/>
    )
}

export default Icon;