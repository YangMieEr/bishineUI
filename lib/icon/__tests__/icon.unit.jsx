import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../Icon';
import { mount } from 'enzyme';

describe('icon', () => {
    it('first time', () => {
        const json = renderer.create(<Icon name='wechat'/>).toJSON();
        expect(json).toMatchSnapshot();
    })

    it('onclick', () => {
        const fn = jest.fn();
        const component = mount(<Icon name='wechat' onClick={fn}/>);
        component.find('i').simulate('click');
        expect(fn).toBeCalled();
    });
})