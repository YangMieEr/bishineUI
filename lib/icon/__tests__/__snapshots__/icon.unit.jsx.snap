// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon first time 1`] = `
<i
  className="iconfont iconwechat"
/>
`;
